core/embed/io/usb/unix/usb.c

```diff
@@ -216,7 +216,7 @@ static int usb_emulated_read(usb_iface_t *iface, uint8_t *buf, uint32_t len) {
     return len;
   }
 
-  return 0;
+  return -1;
```

Comment:
This one open another can of worms i believe. This zero return means that there are no data to read which is 'retriable' error as discussed in previous PR. Furthermore, iiuc stm implementation also returns 0 in some cases when there is nothing to read. STM also seems to return zero if the buffer len is shorter than amount of data to read, which seems a bit weird. Anyway, i'd skip this one for now, seems like it would need a bit more digging.